component/src/test/java/org/wso2/extension/siddhi/io/websocket/util/WebSocketEndpoint.java

```diff
@@ -30,63 +31,44 @@
 import javax.websocket.OnError;
 import javax.websocket.OnMessage;
 import javax.websocket.OnOpen;
-import javax.websocket.Session;
 import javax.websocket.server.PathParam;
 import javax.websocket.server.ServerEndpoint;
 
 @ServerEndpoint(value = "/chat/{name}")
 public class WebSocketEndpoint {
     private static final Logger LOGGER = LoggerFactory.getLogger(WebSocketEndpoint.class);
-    private List<Session> sessions = new LinkedList<Session>();
+    private List<WebSocketConnection> webSocketConnectionList = new LinkedList<>();
 
     @OnOpen
-    public void onOpen(@PathParam("name") String name, Session session) {
-        sessions.add(session);
+    public void onOpen(@PathParam("name") String name, WebSocketConnection websocketConnection) {
+        webSocketConnectionList.add(websocketConnection);
     }
 
     @OnMessage
-    public void onTextMessage(@PathParam("name") String name, String text, Session session) throws IOException {
-        sendMessageToAll(text);
+    public void onTextMessage(@PathParam("name") String name, String text, WebSocketConnection websocketConnection)
+            throws IOException {
+        webSocketConnectionList.forEach(
+                webSocketConnection -> webSocketConnection.pushText(text)
+        );
     }
 
     @OnMessage
-    public void onBinaryMessage(@PathParam("name") String name, ByteBuffer message, Session session) {
-        sendBinaryMessageToAll(message);
-
+    public void onBinaryMessage(@PathParam("name") String name, ByteBuffer message,
+                                WebSocketConnection websocketConnection) {
+        webSocketConnectionList.forEach(
+                webSocketConnection -> webSocketConnection.pushBinary(message)
+        );
     }
 
     @OnClose
-    public void onClose(@PathParam("name") String name, CloseReason closeReason, Session session) {
-        sessions.remove(session);
+    public void onClose(@PathParam("name") String name, CloseReason closeReason,
+                        WebSocketConnection websocketConnection) {
+        webSocketConnectionList.remove(websocketConnection);
```

Comment:
I think we need to remove in the `onError` resource as well.

Reply:
Fixed in a9ca288